python/graphstorm/gconstruct/utils.py

```diff
@@ -378,8 +378,15 @@ def __call__(self, arrs, name):
         if len(arrs) > 1:
             return _merge_arrs(arrs, tensor_path)
         else:
+            # To get the output dtype by accessing the
+            # first element of the arrays (numpy array or HDFArray)
+            # Note: We use arrs[0][0] instead of arrs[0] because
+            #       arrs[0][0] is a transformed data with out_dtype
+            #       while arrs[0] can be a HDFArray and has not
+            #       been cast to out_dtype.
+            dtype = arrs[0][0].dtype
```

Comment:
how many places do you need to change it like this?

Reply:
maybe write a wrapper function?

Reply:
only 2

Reply:
I agree, we need a wrapper function. This part of the code is tricky.